numpyro/mcmc.py

```diff
@@ -269,3 +280,42 @@ def sample_kernel(hmc_state):
         hmc.sample_kernel = sample_kernel
 
     return init_kernel, sample_kernel
+
+
+def mcmc(num_warmup, num_samples, init_samples, sampler='hmc',
+         constrain_fn=None, print_summary=True, **sampler_kwargs):
+    """
+    Convenience wrapper for MCMC samplers -- runs warmup, prints
+    diagnostic summary and returns a collections of samples
+    from the posterior.
+
+    :param num_warmup: Number of warmup steps.
+    :param num_samples: Number of samples to generate from the Markov chain.
+    :param init_samples: Initial parameters to begin sampling. The type can
+        must be consistent with the input type to `potential_fn`.
+    :param sampler: currently, only `hmc` is implemented (default).
+    :param constrain_fn: Callable that converts a collection of unconstrained
+        sample values returned from the sampler to constrained values that
+        lie within the support of the sample sites.
+    :param print_summary: Whether to print diagnostics summary for
+        each sample site. Default is ``True``.
+    :param `**sampler_kwargs`: Sampler specific keyword arguments.
+    :return: collection of samples from the posterior.
+    """
+    if sampler == 'hmc':
+        potential_fn = sampler_kwargs.pop('potential_fn')
+        kinetic_fn = sampler_kwargs.pop('kinetic_fn', None)
+        algo = sampler_kwargs.pop('algo', 'NUTS')
+        progbar = sampler_kwargs.pop('progbar', True)
+
+        init_kernel, sample_kernel = hmc(potential_fn, kinetic_fn, algo)
+        hmc_state = init_kernel(init_samples, num_warmup, progbar=progbar, **sampler_kwargs)
+        hmc_states = fori_collect(num_samples, sample_kernel, hmc_state,
+                                  transform=lambda x: constrain_fn(x.z),
```

Comment:
We should set default `constrain_fn` value, otherwise, this line will trigger an error when it is None